raiden/transfer/node.py

```diff
@@ -600,7 +601,7 @@ def handle_node_change_network_state(
     network_state = state_change.network_state
     chain_state.nodeaddresses_to_networkstates[node_address] = network_state
 
-    for secrethash, subtask in chain_state.payment_mapping.secrethashes_to_task.items():
+    for secrethash, subtask in list(chain_state.payment_mapping.secrethashes_to_task.items()):
```

Comment:
Why do you wrap this with `list(...)`?

Reply:
we need a copy of the iterated sequence -- `subdispatch_mediatortask` can in some instances manipulate the `secrethashes_to_task` mapping.